File: graph/arch/1584-min-cost-connect-points/golang/min-cost-connect-points.go
package main

type point []int

func makePoint(x, y int) point {
	return []int{x, y}
}

func (p1 point) distance(p2 point) int {
	var d int
	if p1[0] < p2[0] {
		d = p2[0] - p1[0]
	} else {
		d = p1[0] - p2[0]
	}
	if p1[1] < p2[1] {
		d = d + p2[1] - p1[1]
	} else {
		d = d + p1[1] - p2[1]
	}
	return d
}

func edges(vs [][]int) (xs []Edge) {
	n := len(vs)
	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			w := point(vs[i]).distance(point(vs[j]))
			xs = append(xs, Edge{u: i, v: j, w: w})
		}
	}
	return
}

File: graph/arch/1584-min-cost-connect-points/py/min_cost_connect_points.py
from typing import List, Tuple
from graph import Edge

Point = Tuple[int, int]


def edges(pts: List[Point]) -> List[Edge]:
  n = len(pts)
  xs = []
  for i in range(n):
    for j in range(i+1, n):
      xs.append([i, j, distance(pts[i], pts[j])])
  return xs


def distance(u, v: Point) -> int:
  return abs(u[0]-v[0]) + abs(u[1]-v[1])
